airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -185,13 +185,13 @@ public void run(final ConnectionUpdaterInput connectionUpdaterInput) throws Retr
       // The workflow state will be updated to true if a reset happened while a job was running.
       // We need to propagate that to the new run that will be continued as new.
       if (workflowState.isResetConnection() && !workflowState.isCancelled()) {
-        connectionUpdaterInput.setResetConnection(true);
+        workflowState.setContinueAsReset(true);
         connectionUpdaterInput.setJobId(null);
         connectionUpdaterInput.setAttemptNumber(1);
         connectionUpdaterInput.setFromFailure(false);
         connectionUpdaterInput.setAttemptId(null);
       } else {
-        connectionUpdaterInput.setResetConnection(false);
+        workflowState.setContinueAsReset(false);
```

Comment:
Can you update the comment to describe this case?

Reply:
Done + explain why we are using the workflow state here